### ISSUE TITLE
`line` property of `no-empty-source` warnings are `undefined`

### ISSUE THREAD

**Issue by shinnn** [styled-components/stylelint-processor-styled-components#178]
### Problem

```js
const {lint} = require('stylelint'); // v9.1.3

(async () => {
  const {results} = await lint({
    code: '',
    config: {
      rules: {'no-empty-source': true}
    }
  });

  console.log(results[0].warnings[0]);
})();
```

When I run the code above, it prints

```js
{ line: 1,
  column: 1,
  rule: 'no-empty-source',
  severity: 'error',
  text: 'Unexpected empty source (no-empty-source)' }
```

But when I use `stylelint-processor-styled-components`,

```js
const {lint} = require('stylelint'); // v9.1.3

(async () => {
  const {results} = await lint({
    code: '',
    config: {
      processors: ['stylelint-processor-styled-components'], // v1.3.1
      rules: {'no-empty-source': true}
    }
  });

  console.log(results[0].warnings[0]);
})();
```

it prints

```js
{ line: undefined,
  column: 1,
  rule: 'no-empty-source',
  severity: 'error',
  text: 'Unexpected empty source (no-empty-source)' }
```

As you can see, this processor makes `line` property of the `no-empty-source` warning `undefined`.

### Expected result

In this case, I expect `line` to be `1` even if stylelint-processor-styled-components is enabled.

**Comment by emilgoldsmith** [comment]
Hmm, well for one we recommend using our shared config that disables warnings that don't make sense with our processor (such as that one), but it seems you have some very specific use case you want to use here. It would probably be a pretty simple fix, I guess it's because there aren't any lines so we aren't building our sourcemaps and then `sourceMaps[1] = undefined`. If you want to look into this and submit a PR feel free, but I don't think this is high priority for us

**Comment by emilgoldsmith** [comment]
Oh okay, if it has further repercussions it could definitely something worth fixing, sorry, I just didn't see the usecase, we could even do something as simple as just outputting a default value of 1 as the line number
